src/profile-logic/profile-data.js

```diff
@@ -978,20 +979,45 @@ export function getTracingMarkers(thread: Thread): TracingMarker[] {
       };
       tracingMarkers.push(marker);
     } else if (data.type === 'tracing') {
+      // Tracing markers come in the form of 2 distinct markers: the first is
+      // emitted at the start of some process, the second at the end. We convert
+      // these 2 one-time markers into one marker with a non-null duration.
+      // We also handle nested markers by assuming markers of the same type are
+      // never interwoven: given input markers startA, startB, endC, endD, we'll
+      // get 2 markers A-D and B-C.
+      // This is implemented by AutoProfilerTracing in Gecko.
```

Comment:
I got confused by the use of numbers (like `2 one-time`), and the word `process`. How about this phrasing?

```
// Tracing markers are created from two distinct markers that are created at
// the start and end of whatever code that is running that we care about. We
// convert both of these raw markers into a single tracing marker with a
// non-null duration. This is implemented by AutoProfilerTracing in Gecko.
//
// We also handle nested markers by assuming markers of the same type are
// never interwoven: given input markers startA, startB, endC, endD, we'll
// get 2 markers A-D and B-C.
```

Reply:
This is much better, thanks

Reply:
I just moved around some sentences. Here is what I used:
```
      // Tracing markers are created from two distinct markers that are created at
      // the start and end of whatever code that is running that we care about.
      // This is implemented by AutoProfilerTracing in Gecko.
      //
      // In this function we convert both of these raw markers into a single
      // tracing marker with a non-null duration.
      //
      // We also handle nested markers by assuming markers of the same type are
      // never interwoven: given input markers startA, startB, endC, endD, we'll
      // get 2 markers A-D and B-C.
```